http-clients/src/main/java/org/triplea/http/client/web/socket/WebSocketConnection.java

```diff
@@ -77,12 +78,18 @@ void close() {
    * Initiates a websocket connection. Must be called before {@link #sendMessage(String)}. This
    * method blocks until the connection has either successfully been established, or the connection
    * failed.
+   *
+   * @throws RuntimeException If the connection fails
    */
   void connect() {
     Preconditions.checkState(!client.isOpen());
     Preconditions.checkState(!closed);
-    Interruptibles.await(
-        () -> client.connectBlocking(DEFAULT_CONNECT_TIMEOUT_MILLIS, TimeUnit.MILLISECONDS));
+    final Result<Boolean> result =
+        Interruptibles.awaitResult(
+            () -> client.connectBlocking(DEFAULT_CONNECT_TIMEOUT_MILLIS, TimeUnit.MILLISECONDS));
+    if (!result.completed || !result.result.orElse(false)) {
+      throw new RuntimeException("Failed to connect to a websocket at " + client.getURI());
```

Comment:
Instead of throwing RuntimeException, it might be good to resurrect the `CouldNotConnect` exception from `WebSocketConnector`:
```
  /** Exception indicating connection to server failed. */
  @VisibleForTesting
  static final class CouldNotConnect extends RuntimeException {
    private static final long serialVersionUID = -5403199291005160495L;

    private static final String ERROR_MESSAGE = "Error, could not connect to server";

    CouldNotConnect() {
      super(ERROR_MESSAGE);
    }

    CouldNotConnect(final Exception e) {
      super(ERROR_MESSAGE, e);
    }
  }
```


I note this as the audience for this error message is the average game player, not developers. A game player will not necessarily know about a websocket, what it's used for, or will they really even care. They'll want to know what the error means to them and the game.  

I don't know if this is that easy to fix. We probably should ideally surface this as a boolean return value and allow the client code decide how to message the failure. The only problem is that in turn, `GenericWebsocketClient` should do the same thing, but we have an async connect which makes it tricky.

There can be a number of solutions, if of any help, I can suggest two courses of action:
1) Surface a boolean return value to `GenericWebSocketClient`, and pass an error message factory to `GenericWebsocketClient` to display on case of failure to connect.
2) Perhaps simpler, just fix up the error message. I think it's debatable if we need to report the URI to the client (could be noise, the URI is completely transparent to users, so it might not make sense to report it). could be something like:
```
throw new  CouldNotConnect()
```
or:
```
throw new CouldNotConnect(client.getURI());
:
:
    CouldNotConnect(URI uri) {
      super(ERROR_MESSAGE + " at " + uri);
    }
```